controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController


joint_trajectory_controller:
  ros__parameters:
    joints:
      - base_rot1
      - rot1_rot2
      - rot2_rot3
      - rot3_prism1
      - prism1_rot4

    command_interfaces:
      - position

    state_interfaces:
      - position
      - velocity

    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    open_loop_control: true
    allow_integration_in_goal_trajectories: true

    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0
      base_rot1:
        trajectory: 0.05
        goal: 0.03
      rot1_rot2:
        trajectory: 0.05
        goal: 0.03
      rot2_rot3:
        trajectory: 0.05
        goal: 0.03
      rot3_prism1:
        trajectory: 0.01
        goal: 0.005
      prism1_rot4:
        trajectory: 0.05
        goal: 0.03
